BraveShared/Rewards/MonthlyAdsGrantReminder.swift

```diff
@@ -0,0 +1,103 @@
+    static public func isMonthlyAdsReminderNotification(_ notification: UNNotification) -> Bool {
+        return notification.request.identifier.hasPrefix(idPrefix)
+    }
+    
+    /// Get the identifier for a notification given its month
+    static private func identifier(for month: Int) -> String {
+        return "\(idPrefix)-\(month)"
+    }
+    
+    /// Cancels the current month's notification if one exists
+    ///
+    /// Trigger this when the user claims an ad grant.
+    static public func cancelCurrentMonth() {
+        let month = Calendar.current.component(.month, from: Date())
+        let id = identifier(for: month)
+        let center = UNUserNotificationCenter.current()
+        center.removePendingNotificationRequests(withIdentifiers: [id])
+        center.removeDeliveredNotifications(withIdentifiers: [id])
+        log.debug("Cancelled monthly ad grant reminder for month: \(month)")
+    }
+    
+    /// Retrieve the next month as an integer (1-12). For example, if the current month was december,
+    /// this would return `1` for January.
+    static private func nextMonth() -> Int? {
+        guard let nextMonthsDate = Calendar.current.date(byAdding: .month, value: 1, to: Date()) else {
+            // Apocalypse...
+            return nil
+        }
+        return Calendar.current.component(.month, from: nextMonthsDate)
+    }
+    
+    /// Schedules a notification for the following month if one doesn't already exist for that month
+    ///
+    /// Trigger this when the user views an ad
+    static public func schedule() {
+        guard let month = nextMonth() else {
+            log.error("Failed to obtain month to schedule notification")
+            return
+        }
+        let center = UNUserNotificationCenter.current()
+        center.requestAuthorization(options: [.provisional, .alert, .sound, .badge]) { (granted, error) in
+            if let error = error {
+                log.error("Failed to request notifications permissions: \(error)")
+                return
+            }
+            if !granted {
+                log.info("Not authorized to schedule a notification")
+                return
+            }
+            
+            let id = self.identifier(for: month)
+            
+            center.getPendingNotificationRequests { requests in
+                if requests.contains(where: { $0.identifier == id }) {
+                    // Already has one scheduled no need to schedule again
+                    return
+                }
+                
+                let content = UNMutableNotificationContent()
+                content.title = Strings.MonthlyAdsClaimNotificationTitle
+                content.body = Strings.MonthlyAdsClaimNotificationBody
+                
+                let trigger = UNCalendarNotificationTrigger(
+                    dateMatching: .init(month: month, day: 7, hour: 13, minute: 21),
```

Comment:
Why this date precisely? Will it work correctly for 12-hour formats with the `hour = 13`?

Reply:
@iccub Basically just was asked for that to be the time it goes off in the ticket (1:21pm or 13:21 in 24-hour clocks). But yes, it definitely works with 12 hour formats, tested on my device which is in 12-hour mode